src/components/SpectralProfiler/SpectralProfilerComponent.tsx

```diff
@@ -62,7 +66,51 @@ export class SpectralProfilerComponent extends React.Component<WidgetProps> {
         return 20 + (this.widgetStore.settingsPanelVisible ? PANEL_CONTENT_WIDTH : 0);
     }
 
-    @computed get plotData(): { values: Array<Point2D>, xMin: number, xMax: number, yMin: number, yMax: number, yMean: number, yRms: number } {
+    // Qi, calculate PA according datas
+    private calculatePA(datas: Array<any>): Array<number> {
+        let vals = [];
+        if (datas[0] && datas[1] && datas.length === 2 && datas[0].length === datas[1].length) {
+            for (let i = 0; i < datas[0].vals.length; i++) {
+                vals[i] = Math.atan(datas[1].vals[i] / datas[0].vals[i]);
+            }
+        }
+        return vals;
+    }
+
+    // Qi, calculate PI according datas
+    private calculatePI(datas: Array<any>): Array<number> {
+        let vals = [];
+        if (datas[0] && datas[1] && datas.length === 2 && datas[0].length === datas[1].length) {
+            for (let i = 0; i < datas[0].vals.length; i++) {
+                vals[i] = Math.sqrt(Math.pow(datas[0].vals[i], 2) + Math.pow(datas[1].vals[i], 2));
+            }
+        }
+        return vals;
+    }
+
+    // Qi, return PI, PA data for ploting
+    private calculate(coordinate: string, statsType: CARTA.StatsType): CARTA.ISpectralProfile {
+        let coordinateData: CARTA.ISpectralProfile;
+        if (coordinate === "PIz" || coordinate === "PAz") {
```

Comment:
I think we should define some of these strings as constants. Perhaps in an enum? I'd rather see something like
````
if (coordinate === StokesCoordinate.PolarizedIntensity || coordiante === StokesCoordinate.PolarizationAngle)
````